File: 리트코드/0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
    
        return binarySearch(nums, target, 0, nums.length - 1);
    
    }

    public int binarySearch(int[] nums, int target, int left, int right){
        if(left <= right){
            int mid = (left + right)/2;
            if(nums[mid] > target){
                return binarySearch(nums,target, left ,mid - 1);
            }else if(nums[mid] < target){
                return binarySearch(nums,target, mid + 1,right);
            }else{
                return mid;
            }
        } else{
            return -1;
        }
    }
}

File: 리트코드/0704-binary-search/0704-binary-search.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        fun binarySearch(left:Int, right:Int): Int{
            if(left <= right){
                val mid = (left + right)/2
                if(nums[mid] > target){
                    return binarySearch(left, mid-1)
                }else if(nums[mid] < target){
                    return binarySearch(mid+1, right)
                }else{
                    return mid
                }
            }
            return -1
        }

        return binarySearch(0,nums.size - 1)
    }
}